### PR TITLE
Improve Contribution Guide's VSCode setup instructions 

### PR DESCRIPTION
**Pull request by evpirnia**
###  Issue 
While following the instructions for Setting Up Your Environment using Visual Studio Code, I was prompted with 3 settings options when the docs imply that just one option would be presented. I think the docs need to be updated to include info on the 3 different options and link to the VSCode docs for the user to identify and select which one would work best for their use case. See https://github.com/fastify/fastify/issues/4593 

### Fix 

Update Contribution Guide doc to include info on the 3 different options available now for VSCode Settings. Add link to VSCode doc for the user to identify and select which one would work best for their use case. 

### Tests 
```
» npm run test

> fastify@4.13.0 test
> npm run lint && npm run unit && npm run test:typescript


> fastify@4.13.0 lint
> npm run lint:standard && npm run lint:typescript && npm run lint:markdown


> fastify@4.13.0 lint:standard
> standard | snazzy


> fastify@4.13.0 lint:typescript
> eslint -c types/.eslintrc.json types/**/*.d.ts test/types/**/*.test-d.ts


> fastify@4.13.0 lint:markdown
> markdownlint-cli2

markdownlint-cli2 v0.6.0 (markdownlint v0.27.0)

> fastify@4.13.0 unit
> tap

​ PASS ​ test/build/version.test.js 1 OK 10.612ms
​ PASS ​ test/als.test.js 13 OK 98.524ms
​ PASS ​ test/500s.test.js 29 OK 91.35ms
​ PASS ​ test/bodyLimit.test.js 17 OK 282.42ms
​ PASS ​ test/allowUnsafeRegex.test.js 14 OK 297.524ms
​ PASS ​ test/404s.test.js 297 OK 965.623ms
​ PASS ​ test/chainable.test.js 3 OK 17.949ms
​ PASS ​ test/async-await.test.js 81 OK 1s
​ PASS ​ test/case-insensitive.test.js 18 OK 156.327ms
​ PASS ​ test/client-timeout.test.js 1 OK 44.792ms
​ SKIP ​ test/close-pipelining.test.js
 ~ Should close the socket abruptly - pipelining - return503OnClosing: false, skip Node < v19.x

​ SKIP ​ test/close-pipelining.test.js 1 skip of 3 138.159ms
 ~ Should close the socket abruptly - pipelining - return503OnClosing: false, skip Node < v19.x

​ SKIP ​ test/close.test.js
 ~ Current opened connection should NOT continue to work after closing and return "connection: close" header - return503OnClosing: false, skip Node < v19.x

​ SKIP ​ test/build/error-serializer.test.js
 ~ ensure the current error serializer is latest

​ SKIP ​ test/close.test.js
 ~ shutsdown while keep-alive connections are active (non-async, native)

​ SKIP ​ test/close.test.js
 ~ shutsdown while keep-alive connections are active (non-async, idle, native)

​ SKIP ​ test/build/error-serializer.test.js 1 skip of 2 2s
 ~ ensure the current error serializer is latest

​ SKIP ​ test/close.test.js 3 skip of 51 580.716ms
 ~ Current opened connection should NOT continue to work after closing and return "connection: close" header - return503OnClosing: false, skip Node < v19.x
 ~ shutsdown while keep-alive connections are active (non-async, native)
 ~ shutsdown while keep-alive connections are active (non-async, idle, native)

​ PASS ​ test/connectionTimeout.test.js 6 OK 37.672ms
​ PASS ​ test/content-type.test.js 8 OK 64.993ms
​ PASS ​ test/content-length.test.js 22 OK 101.931ms
​ PASS ​ test/constrained-routes.test.js 76 OK 171.93ms
​ PASS ​ test/content-parser.test.js 95 OK 300.948ms
​ PASS ​ test/context-config.test.js 27 OK 75.209ms
​ PASS ​ test/custom-http-server.test.js 6 OK 77.547ms
​ PASS ​ test/copy.test.js 4 OK 98.305ms
​ PASS ​ test/custom-parser-async.test.js 7 OK 130.375ms
​ PASS ​ test/custom-querystring-parser.test.js 28 OK 168.231ms
​ PASS ​ test/default-route.test.js 8 OK 55.239ms
​ PASS ​ test/delete.test.js 43 OK 458.538ms
​ PASS ​ test/encapsulated-error-handler.test.js 7 OK 64.919ms
​ PASS ​ test/esm/index.test.js 2 OK 31.156ms
​ PASS ​ test/diagnostics-channel.test.js 5 OK 675.227ms
​ PASS ​ test/decorator.test.js 193 OK 867.645ms
​ PASS ​ test/fastify-instance.test.js 12 OK 50.318ms
​ PASS ​ test/custom-parser.test.js 224 OK 1s
​ PASS ​ test/fluent-schema.test.js 18 OK 241.051ms
​ PASS ​ test/genReqId.test.js 4 OK 60.125ms
​ PASS ​ test/has-route.test.js 8 OK 32.525ms
​ PASS ​ test/handler-context.test.js 9 OK 54.536ms
​ PASS ​ test/get.test.js 53 OK 381.488ms
​ PASS ​ test/head.test.js 17 OK 243.368ms
​ PASS ​ test/hooks-async.test.js 112 OK 331.486ms
​ PASS ​ test/hooks.on-ready.test.js 75 OK 671.189ms
​ PASS ​ test/http2/head.test.js 3 OK 166.388ms
​ PASS ​ test/http2/closing.test.js 9 OK 344.113ms
​ PASS ​ test/http2/missing-http2-module.test.js 2 OK 15.233ms
​ PASS ​ test/http2/plain.test.js 6 OK 133.004ms
​ PASS ​ test/http2/constraint.test.js 9 OK 231.258ms
​ PASS ​ test/http2/secure-with-fallback.test.js 14 OK 226.836ms
​ PASS ​ test/http2/unknown-http-method.test.js 3 OK 110.168ms
​ PASS ​ test/http2/secure.test.js 7 OK 171.391ms
​ PASS ​ test/imports.test.js 4 OK 437.174ms
​ PASS ​ test/https/custom-https-server.test.js 5 OK 101.917ms
​ PASS ​ test/https/https.test.js 10 OK 187.163ms
​ PASS ​ test/internals/decorator.test.js 17 OK 37.323ms
​ PASS ​ test/internals/contentTypeParser.test.js 4 OK 47.966ms
​ PASS ​ test/internals/all.test.js 32 OK 78.236ms
​ PASS ​ test/internals/hookRunner.test.js 89 OK 60.882ms
​ PASS ​ test/internals/hooks.test.js 27 OK 33.664ms
​ PASS ​ test/inject.test.js 66 OK 1s
​ PASS ​ test/internals/handleRequest.test.js 32 OK 408.942ms
​ PASS ​ test/internals/plugin.test.js 9 OK 54.044ms
​ PASS ​ test/internals/logger.test.js 40 OK 94.597ms
​ PASS ​ test/internals/request.test.js 80 OK 62.254ms
​ PASS ​ test/internals/initialConfig.test.js 67 OK 223.351ms
​ PASS ​ test/internals/reply-serialize.test.js 103 OK 229.894ms
​ PASS ​ test/internals/server.test.js 8 OK 116.342ms
​ PASS ​ test/internals/validation.test.js 33 OK 109.993ms
​ PASS ​ test/keepAliveTimeout.test.js 6 OK 47.1ms
​ PASS ​ test/internals/request-validate.test.js 216 OK 731.24ms
​ PASS ​ test/listen.deprecated.test.js 29 OK 206.887ms
​ PASS ​ test/internals/reply.test.js 314 OK 2s
​ PASS ​ test/listen.test.js 53 OK 329.466ms
​ PASS ​ test/lock.test.js 5 OK 112.327ms
​ PASS ​ test/noop-set.test.js 2 OK 8.105ms
​ PASS ​ test/hooks.test.js 588 OK 6s
​ PASS ​ test/middleware.test.js 3 OK 36.97ms
​ PASS ​ test/maxRequestsPerSocket.test.js 22 OK 134.784ms
​ PASS ​ test/logger.test.js 281 OK 691.421ms
​ PASS ​ test/mkcol.test.js 4 OK 113.889ms
​ PASS ​ test/move.test.js 5 OK 121.423ms
​ PASS ​ test/nullable-validation.test.js 18 OK 261.093ms
​ PASS ​ test/output-validation.test.js 24 OK 268.376ms
​ PASS ​ test/pretty-print.test.js 24 OK 82.323ms
​ PASS ​ test/options.error-handler.test.js 79 OK 2s
​ PASS ​ test/proppatch.test.js 5 OK 191.345ms
​ PASS ​ test/plugin.test.js 221 OK 1s
​ PASS ​ test/promises.test.js 20 OK 634.595ms
​ PASS ​ test/propfind.test.js 11 OK 311.38ms
​ PASS ​ test/options.test.js 69 OK 2s
​ PASS ​ test/patch.error-handler.test.js 104 OK 2s
​ PASS ​ test/patch.test.js 80 OK 2s
​ PASS ​ test/proto-poisoning.test.js 21 OK 161.599ms
​ PASS ​ test/reply-code.test.js 8 OK 63.887ms
​ PASS ​ test/register.test.js 31 OK 149.43ms
test/reply-trailers.test.js 2> (node:57556) [FSTDEP013] FastifyDeprecation: Direct return of "trailers" function is deprecated. Please use "callback" or
"async-await" for return value. The support of direct return will removed in `fastify@5`.
test/reply-trailers.test.js 2> (Use `node --trace-warnings ...` to show where the warning was created)
​ PASS ​ test/request-error.test.js 52 OK 324.223ms
​ PASS ​ test/reply-error.test.js 301 OK 694.587ms
​ PASS ​ test/requestTimeout.test.js 7 OK 38.288ms
​ PASS ​ test/put.test.js 80 OK 1s
​ PASS ​ test/put.error-handler.test.js 104 OK 1s
​ PASS ​ test/route-prefix.test.js 89 OK 227.628ms
​ PASS ​ test/reply-trailers.test.js 110 OK 1s
​ PASS ​ test/router-options.test.js 35 OK 149.847ms
​ PASS ​ test/route-hooks.test.js 210 OK 859.708ms
​ PASS ​ test/same-shape.test.js 4 OK 71.381ms
test/route.test.js 2> (node:57562) [FSTDEP012] FastifyDeprecation: Request#context property access is deprecated. Please use "Request#routeConfig" or
"Request#routeSchema" instead for accessing Route settings. The "Request#context" will be removed in `fastify@5`.
test/route.test.js 2> (Use `node --trace-warnings ...` to show where the warning was created)
​ PASS ​ test/route.test.js 185 OK 634.891ms
​ PASS ​ test/schema-serialization.test.js 102 OK 420.329ms
​ PASS ​ test/schema-examples.test.js 23 OK 597.48ms
​ PASS ​ test/schema-feature.test.js 185 OK 1s
​ PASS ​ test/server.test.js 5 OK 83.379ms
​ PASS ​ test/schema-special-usage.test.js 50 OK 671.14ms
​ PASS ​ test/search.test.js 31 OK 327.794ms
​ PASS ​ test/schema-validation.test.js 100 OK 646.159ms
​ PASS ​ test/sync-routes.test.js 5 OK 64.62ms
​ PASS ​ test/trace.test.js 1 OK 11.985ms
​ SKIP ​ test/stream.test.js
 ~ should support send module 200 and 404

​ PASS ​ test/throw.test.js 38 OK 171.93ms
​ PASS ​ test/type-provider.test.js 2 OK 21.86ms
​ PASS ​ test/trust-proxy.test.js 35 OK 284.292ms
​ SKIP ​ test/stream.test.js 1 skip of 85 705.784ms
 ~ should support send module 200 and 404

​ PASS ​ test/unlock.test.js 4 OK 111.042ms
​ PASS ​ test/unsupported-httpversion.test.js 2 OK 63.122ms
​ PASS ​ test/wrapThenable.test.js 2 OK 18.992ms
​ PASS ​ test/upgrade.test.js 2 OK 58.573ms
​ PASS ​ test/url-rewriting.test.js 14 OK 150.748ms
​ PASS ​ test/validation-error-handling.test.js 86 OK 394.618ms
​ PASS ​ test/versioned-routes.test.js 89 OK 361.379ms
​ PASS ​ test/skip-reply-send.test.js 158 OK 10s
​

                         
  🌈 SUMMARY RESULTS 🌈  
                         
​ SKIP ​ test/build/error-serializer.test.js 1 skip of 2 2s
 ~ ensure the current error serializer is latest

​ SKIP ​ test/close-pipelining.test.js 1 skip of 3 138.159ms
 ~ Should close the socket abruptly - pipelining - return503OnClosing: false, skip Node < v19.x

​ SKIP ​ test/close.test.js 3 skip of 51 580.716ms
 ~ Current opened connection should NOT continue to work after closing and return "connection: close" header - return503OnClosing: false, skip Node < v19.x
 ~ shutsdown while keep-alive connections are active (non-async, native)
 ~ shutsdown while keep-alive connections are active (non-async, idle, native)

​ SKIP ​ test/stream.test.js 1 skip of 85 705.784ms
 ~ should support send module 200 and 404

​
Suites:   ​123 passed​, ​123 of 123 completed​
Asserts:  ​​​6711 passed​, ​6 skip​, ​of 6717​
​Time:​   ​31s​
-----------------------------|---------|----------|---------|---------|-------------------
File                         | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
-----------------------------|---------|----------|---------|---------|-------------------
All files                    |     100 |      100 |     100 |     100 |                   
 fastify                     |     100 |      100 |     100 |     100 |                   
  fastify.js                 |     100 |      100 |     100 |     100 |                   
 fastify/lib                 |     100 |      100 |     100 |     100 |                   
  contentTypeParser.js       |     100 |      100 |     100 |     100 |                   
  context.js                 |     100 |      100 |     100 |     100 |                   
  decorate.js                |     100 |      100 |     100 |     100 |                   
  error-handler.js           |     100 |      100 |     100 |     100 |                   
  errors.js                  |     100 |      100 |     100 |     100 |                   
  fourOhFour.js              |     100 |      100 |     100 |     100 |                   
  handleRequest.js           |     100 |      100 |     100 |     100 |                   
  headRoute.js               |     100 |      100 |     100 |     100 |                   
  hooks.js                   |     100 |      100 |     100 |     100 |                   
  httpMethods.js             |     100 |      100 |     100 |     100 |                   
  initialConfigValidation.js |     100 |      100 |     100 |     100 |                   
  logger.js                  |     100 |      100 |     100 |     100 |                   
  noop-set.js                |     100 |      100 |     100 |     100 |                   
  pluginOverride.js          |     100 |      100 |     100 |     100 |                   
  pluginUtils.js             |     100 |      100 |     100 |     100 |                   
  reply.js                   |     100 |      100 |     100 |     100 |                   
  reqIdGenFactory.js         |     100 |      100 |     100 |     100 |                   
  request.js                 |     100 |      100 |     100 |     100 |                   
  route.js                   |     100 |      100 |     100 |     100 |                   
  schema-controller.js       |     100 |      100 |     100 |     100 |                   
  schemas.js                 |     100 |      100 |     100 |     100 |                   
  server.js                  |     100 |      100 |     100 |     100 |                   
  symbols.js                 |     100 |      100 |     100 |     100 |                   
  validation.js              |     100 |      100 |     100 |     100 |                   
  warnings.js                |     100 |      100 |     100 |     100 |                   
  wrapThenable.js            |     100 |      100 |     100 |     100 |                   
-----------------------------|---------|----------|---------|---------|-------------------

> fastify@4.13.0 test:typescript
> tsc test/types/import.ts && tsd
```

### Benchmark 
```
» npm run benchmark   

> fastify@4.13.0 benchmark
> npx concurrently -k -s first "node ./examples/benchmark/simple.js" "npx autocannon -c 100 -d 30 -p 10 localhost:3000/"

[1] Running 30s test @ http://localhost:3000/
[1] 100 connections with 10 pipelining factor
[1] 
[1] 
[1] ┌─────────┬──────┬───────┬───────┬───────┬──────────┬──────────┬─────────┐
[1] │ Stat    │ 2.5% │ 50%   │ 97.5% │ 99%   │ Avg      │ Stdev    │ Max     │
[1] ├─────────┼──────┼───────┼───────┼───────┼──────────┼──────────┼─────────┤
[1] │ Latency │ 9 ms │ 20 ms │ 29 ms │ 31 ms │ 20.18 ms │ 44.03 ms │ 1775 ms │
[1] └─────────┴──────┴───────┴───────┴───────┴──────────┴──────────┴─────────┘
[1] ┌───────────┬─────┬──────┬─────────┬─────────┬──────────┬─────────┬─────────┐
[1] │ Stat      │ 1%  │ 2.5% │ 50%     │ 97.5%   │ Avg      │ Stdev   │ Min     │
[1] ├───────────┼─────┼──────┼─────────┼─────────┼──────────┼─────────┼─────────┤
[1] │ Req/Sec   │ 0   │ 0    │ 50463   │ 59391   │ 49321.07 │ 10059.7 │ 38230   │
[1] ├───────────┼─────┼──────┼─────────┼─────────┼──────────┼─────────┼─────────┤
[1] │ Bytes/Sec │ 0 B │ 0 B  │ 9.49 MB │ 11.2 MB │ 9.27 MB  │ 1.89 MB │ 7.19 MB │
[1] └───────────┴─────┴──────┴─────────┴─────────┴──────────┴─────────┴─────────┘
[1] 
[1] Req/Bytes counts sampled once per second.
[1] # of samples: 30
[1] 
[1] 1481k requests in 30.62s, 278 MB read
[1] npx autocannon -c 100 -d 30 -p 10 localhost:3000/ exited with code 0
--> Sending SIGTERM to other processes..
[0] node ./examples/benchmark/simple.js exited with code SIGTERM
```

### CODE REVIEW

File: docs/Guides/Contributing.md [hunk 1]
```diff
@@ -162,9 +162,11 @@ the left sidebar. But wait! We are not quite done yet. There are a few more
 baseline settings that should be set before VSCode is ready.
 
 Press `cmd+shift+p` to bring up the VSCode command input prompt. Type `open
-settings (json)` and then choose the same item from the filtered menu. This will
-open a document that is the settings for the editor. Paste the following JSON
-into this document, overwriting any text already present, and save it:
+settings (json)`. Three options will appear in the dropdown: Workspace, Default,
+and User settings. Select the [VSCode Setting](https://code.visualstudio.com/docs/getstarted/settings)
+that works best for you. This will open a document that is the settings for the
```
Comment (jsumners) [inline docs/Guides/Contributing.md:167]: Shouldn't we recommend a specific choice that is in line with setting up this Fastify dedicated editor? I think we should direct the reader to choose "Default" in order to edit the global editor settings. This way their Fastify specific editor will work for all repos in our org.

Comment (evpirnia) [reply docs/Guides/Contributing.md:167]: yes, love it! I'll add that now.

**Review by mcollina — APPROVED** [review]
lgtm

**Review by jsumners — APPROVED** [review]
Thank you.

**Pull request merged**